python/pyarrow/tests/util.py

```diff
@@ -178,6 +178,7 @@ def get_modified_env_with_pythonpath():
     else:
         new_pythonpath = module_path
     env['PYTHONPATH'] = new_pythonpath
+    env['MACOSX_DEPLOYMENT_TARGET'] = "14.0"
```

Comment:
The cython tests that were running setup.py in a subprocess and linking against the pyarrow-exposed arrow libraries were failing on macOS without this